Userland/Libraries/LibJS/Runtime/Temporal/ISO8601.cpp

```diff
@@ -876,50 +885,163 @@ bool ISO8601Parser::parse_time_zone()
+{
+    // AnnotationKeyTail :
+    //     AKeyChar AnnotationKeyTail[opt]
+    if (!parse_a_key_char())
+        return false;
+    // This is implemented without recursion to prevent stack overflow with annotation key tails that have many characters.
+    while (parse_a_key_char())
+        ;
+    return true;
+}
+
+// https://tc39.es/proposal-temporal/#prod-AnnotationKey
+Optional<StringView> ISO8601Parser::parse_annotation_key()
+{
+    // AnnotationKey :
+    //     AKeyLeadingChar AnnotationKeyTail[opt]
     StateTransaction transaction { *this };
-    do {
-        if (!parse_calendar_name_component())
-            return false;
-    } while (m_state.lexer.consume_specific('-'));
-    m_state.parse_result.calendar_name = transaction.parsed_string_view();
+    if (!parse_a_key_leading_char())
+        return {};
+    (void)parse_annotation_key_tail();
+    auto annotation_key = transaction.parsed_string_view();
     transaction.commit();
+    return annotation_key;
+}
+
+// https://tc39.es/proposal-temporal/#prod-AnnotationValueComponent
+bool ISO8601Parser::parse_annotation_value_component()
+{
+    // AnnotationValueComponent :
+    //     AValChar AnnotationValueComponent[opt]
+    if (!parse_a_val_char())
+        return false;
+    // This is implemented without recursion to prevent stack overflow with annotation value components that have many characters.
+    while (parse_a_val_char())
+        ;
     return true;
 }
 
-// https://tc39.es/proposal-temporal/#prod-Calendar
-bool ISO8601Parser::parse_calendar()
+// https://tc39.es/proposal-temporal/#prod-AnnotationValueTail
+Optional<StringView> ISO8601Parser::parse_annotation_value_tail()
 {
-    // Calendar :
-    //     [u-ca= CalendarName ]
+    // AnnotationValueTail :
+    //     AnnotationValueComponent
+    //     AnnotationValueComponent - AnnotationValueTail
     StateTransaction transaction { *this };
-    if (!m_state.lexer.consume_specific("[u-ca="sv))
+
+    // This is implemented without recursion to prevent stack overflow with annotation values that have many dashes.
+    for (;;) {
+        if (!parse_annotation_value_component())
+            return {};
+
+        if (!m_state.lexer.consume_specific('-'))
+            break;
+    }
+
+    auto annotation_value_tail = transaction.parsed_string_view();
+    transaction.commit();
+    return annotation_value_tail;
+}
+
+// https://tc39.es/proposal-temporal/#prod-AnnotationValue
+Optional<StringView> ISO8601Parser::parse_annotation_value()
+{
+    // AnnotationValue :
+    //     AnnotationValueTail
+    return parse_annotation_value_tail();
+}
+
+bool ISO8601Parser::parse_annotation_value_for_production_parser()
+{
+    return parse_annotation_value().has_value();
+}
+
+// https://tc39.es/proposal-temporal/#prod-Annotation
+bool ISO8601Parser::parse_annotation()
+{
+    // Annotation :
+    //     [ AnnotationCriticalFlag[opt] AnnotationKey = AnnotationValue ]
+    if (!m_state.lexer.consume_specific('['))
```

Comment:
You need to use a transaction here to reset the parse offset in the case of a failure